src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -165,6 +173,22 @@ class Simplex_tree {
     }
   };
 
+ private:
+  /** \brief An iterator for an optimized search for the star of a simplex.
+   *
+   * \details It requires the Options::link_nodes_by_label to be true and store two
+   * extra pointers in each node of the simplex tree. The Nodes of same label are
+   * linked in a list.
+   */
+  using Optimized_star_simplex_iterator = Simplex_tree_optimized_star_simplex_iterator<Simplex_tree>;
+  /** \brief Range for an optimized search for the star of a simplex. */
+  using Optimized_star_simplex_range = boost::iterator_range<Optimized_star_simplex_iterator>;
+
+  using Const_boost_iterator = const boost::container::vec_iterator<std::pair<int, Node >*, false>;
+  using Filtered_boost_predicate = std::function<bool(Const_boost_iterator)>;
+  std::function<bool(Const_boost_iterator)> select;
+  using Optimized_star_simplex_filtered_range = boost::filtered_range<decltype(select), Optimized_star_simplex_range>;
```

Comment:
I am trying to think how safe this is. filtered_range does not keep a copy of its range argument, so returning a filtered_range of a local range that is going out of scope is dangerous. However, filtered_range only uses the range to call begin() and end() on it, and stores filtered versions of both. Since Optimized_star_simplex_range is a boost::iterator_range (i.e. a pair of iterators begin/end), extracting just its begin and end and dropping the range should be safe.
I think this deserves a comment in the code, since any small change can easily break this.

Reply:
I added a comment on [d72926c](https://github.com/GUDHI/gudhi-devel/pull/877/commits/d72926c48a5628f7c72d0deb0fadedfb94776bdd)